src/Microsoft.Azure.EventHubs.Processor/AzureStorageCheckpointLeaseManager.cs

```diff
@@ -241,22 +240,26 @@ public async Task<bool> DeleteLeaseStoreAsync()
 
         public async Task<Lease> GetLeaseAsync(string partitionId) // throws URISyntaxException, IOException, StorageException
         {
-    	    AzureBlobLease retval = null;
+            AzureBlobLease retval = null;
 
             CloudBlockBlob leaseBlob = GetBlockBlobReference(partitionId);
 
             if (await leaseBlob.ExistsAsync(null, this.operationContext).ConfigureAwait(false))
-		    {
+            {
                 retval = await DownloadLeaseAsync(partitionId, leaseBlob).ConfigureAwait(false);
-		    }
+            }
 
             return retval;
         }
 
         public IEnumerable<Task<Lease>> GetAllLeases()
         {
             List<Task<Lease>> leaseFutures = new List<Task<Lease>>();
-            IEnumerable<string> partitionIds = this.host.PartitionManager.GetPartitionIdsAsync().Result;
+            IEnumerable<string> partitionIds =
+                Task.Run(async () => await this.host.PartitionManager.GetPartitionIdsAsync().ConfigureAwait(false))
+                    .GetAwaiter()
+                    .GetResult();
```

Comment:
Good catch. Below is same as Task.Run
this.host.PartitionManager.GetPartitionIdsAsync().GetAwaiter().GetResult();

Reply:
![image](https://user-images.githubusercontent.com/8322252/48359902-c0378f80-e67c-11e8-9cf3-7262ec5ee4f4.png)

Reply:
@serkantkaraca  done , also changed to `yield return insted` I think is ok for now if I found more of these async call in a sync context I can create some helper method to Run this keeping in considerations things like the library `Nito.Asyc` https://github.com/StephenCleary/AsyncEx/blob/db32fd5db0d1051e867b36ae039ea13d2c36eb91/src/Nito.AsyncEx.Tasks/SynchronousTaskExtensions.cs#L11